precompiles/collective/src/lib.rs

```diff
@@ -266,6 +267,52 @@ where
 
 		Ok(hash)
 	}
+
+	#[precompile::public("proposals()")]
+	#[precompile::view]
+	fn proposals(handle: &mut impl PrecompileHandle) -> EvmResult<Vec<H256>> {
+		handle.record_cost(RuntimeHelper::<Runtime>::db_read_gas_cost())?;
+
+		let proposals = pallet_collective::Pallet::<Runtime, Instance>::proposals();
+		let proposals: Vec<_> = proposals.into_iter().map(|hash| hash.into()).collect();
```

Comment:
There is no gas cost accounting for this iteration, conversion and collection in a new Vec.
Do we have a way to benchmark precompiles like substrate calls?

Reply:
We do similar stuff in other precompiles, considering it negligible. We should probably benchmark them yes, but I don't know how to do so.

Reply:
Alternatively, we could charge  a DB read if we feel better about it. I have done this in another precompile where I had to do a hash..

Reply:
Note that here it is already a hash, and we're just converting it to H256 (iterated).